### PR TITLE
Use GRPC interceptors instead of explicit context wrappers

### LINKED ISSUE #6035: [Feature]: Implement bearer token auth interceptor
> we should move upgradeContextWithBearerToken from plugin/storage/grpc/shared/grpc_client.go to be implemented as interceptor in pkg/bearertoken

Should be similar to the pkg/tenancy implementation.

Details: https://github.com/jaegertracing/jaeger/pull/6030#issuecomment-2387077052

### PR DESCRIPTION
**Pull request by chahatsagarmain**
## Which problem is this PR solving?
- Resolves #6035 

## Description of the changes
- 

## How was this change tested?
- 

## Checklist
- [ ] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [ ] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: plugin/storage/grpc/factory.go [hunk 1]
```diff
@@ -128,7 +128,7 @@ func (f *Factory) newRemoteStorage(
 		bearertoken.NewUnaryClientInterceptor(),
 	}
 	streamInterceptors := []grpc.StreamClientInterceptor{
-		tenancy.NewClientStreamInterceptor(tenancyMgr),
+		bearertoken.NewStreamClientInterceptor(),
```
Comment (yurishkuro) [inline plugin/storage/grpc/factory.go:131]: good catch. Can you move `tenancyMgr` variable creation down, just before it is used?

File: plugin/storage/grpc/shared/grpc_client.go [hunk 2]
```diff
@@ -117,7 +82,7 @@ func (c *GRPCClient) ArchiveSpanWriter() spanstore.Writer {
 
 // GetTrace takes a traceID and returns a Trace associated with that traceID
 func (c *GRPCClient) GetTrace(ctx context.Context, traceID model.TraceID) (*model.Trace, error) {
-	stream, err := c.readerClient.GetTrace(upgradeContext(ctx), &storage_v1.GetTraceRequest{
```
Comment (yurishkuro) [inline plugin/storage/grpc/shared/grpc_client.go:120]: where is the code that attaches interceptors to the client that would do these upgrades instead?

Comment (chahatsagarmain) [reply plugin/storage/grpc/shared/grpc_client.go:120]: Interceptors are being attached to shared client in `/plugin/storage/grpc.go` https://github.com/jaegertracing/jaeger/blob/6a520a570dd192935aaa01da29a8c9c732f01277/plugin/storage/grpc/factory.go#L162

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
Thanks!

**Pull request merged**